src/CargoBuild/README.md

```diff
@@ -0,0 +1,121 @@
+### CargoBuild Sdk Prototype
+
+### Project Setup
+To use this sdk you will need the following:
+
+1) in your global.json
+```json
+ "msbuild-sdks": {
+    ...,
+    "Microsoft.Build.CargoBuild": "1.0.270-gf406f8eaa0"
+  },
+```
+
+2) foreach rust project a .cargosproj project file at the same level as your cargo.toml file. The project file should import the CargoBuild sdk.
+```xml
+<Project Sdk="Microsoft.Build.CargoBuild">
+  <PropertyGroup>
+    <TargetFramework> your framework version </TargetFramework>
+  </PropertyGroup>
+</Project>
+```
+
+### Usage
+To restore rust dependencies, you can use the following msbuild command:
+```shell
+msbuild /t:Restore
+```
+or 
+```shell
+msbuild /restore
+``` 
+
+To build a rust project, you can use the following msbuild command:
+```shell
+msbuild /t:Build
+```
+
+To clean a rust project, you can use the following msbuild command:
+```shell
+msbuild /p:clean=true
+```
+
+To run a rust project, you can use the following msbuild command:
+```shell
+msbuild /p:run=true
+```
+
+To run cargo tests:
+```shell
+msbuild /p:test=true
+```
+
+For cargo docs
+```shell
+msbuild /p:doc=true
+```
+
+To clear the cargo home cache
+```shell
+msbuild /t:clearcargocache
+```
+
+### Opening .cargoproj files in Visual Studio
+To work with .cargoproj files in Visual Studio, you will need to add the following to your .cargoproj file. 
+Specifically an itemgroup with a None item for the rust source files.
+
+```xml
+<Project Sdk="Microsoft.Build.CargoBuild">
+  <PropertyGroup>
+    <TargetFrameworks> your framework version </TargetFrameworks>
+      <OutputDir>target</OutputDir>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include="src/*.rs" />
```

Comment:
As mentioned above, why not just let the default items pick this up? You can set the language extension to be `.rs`